src/v/datalake/datalake_manager.cc

```diff
@@ -650,4 +667,62 @@ size_t datalake_manager::partitions_with_translation_blocked() const {
     return 0;
 }
 
+void datalake_manager::update_disk_limits() {
+    // the requests
+    auto new_total_size = _scratch_space_size_bytes();
+    if (!_disk_space_manager_enable()) {
+        // when we disable the manager we want to effectively stop enforcing
+        // limits. so make the total allowable size massive, but not so big that
+        // we need to worry about any kind of integer overflow.
+        new_total_size = 50_TiB;
+    }
+
+    const auto new_soft_percent = _scratch_space_soft_limit_size_percent();
+
+    // current settings
+    const auto prev_total = _disk_bytes_reservable_total;
+    const auto prev_soft_limit = _disk_bytes_reservable_soft_limit;
+    const auto prev_available = _core0_disk_bytes_reservable.available_units();
+
+    _disk_bytes_reservable_soft_limit = [&] {
+        auto new_soft_limit_size = static_cast<size_t>(
+          static_cast<double>(new_total_size) * (new_soft_percent / 100.0));
+        // soft limit should be no larger than the total scratch space size
+        return std::min(new_soft_limit_size, new_total_size);
```

Comment:
is this even possible since we have bound checks on `new_soft_percent` configuration.